runtime/doc/develop.txt

```diff
@@ -166,77 +166,282 @@ stick to these rules, to keep the sources readable and maintainable.
 This list is not complete.  Look in the source code for more examples.
 
 
-MAKING CHANGES						*style-changes*
-
-The basic steps to make changes to the code:
-1. Get the code from github.  That makes it easier to keep your changed
-   version in sync with the main code base (it may be a while before your
-   changes will be included).  You do need to spend some time learning git,
-   it's not the most user friendly tool.
-2. Adjust the documentation.  Doing this first gives you an impression of how
-   your changes affect the user.
-3. Make the source code changes.
-4. Check ../doc/todo.txt if the change affects any listed item.
-5. Make a patch with "git diff".  You can also create a pull request on
-   github, but it's the diff that matters.
-6. Make a note about what changed, preferably mentioning the problem and the
-   solution.  Send an email to the |vim-dev| maillist with an explanation and
-   include the diff. Or create a pull request on github.
-
-
 C COMPILER				*style-compiler* *ANSI-C* *C89* *C99*
 
 The minimal C compiler version supported is C89, also known as ANSI C.
 Later standards, such as C99, are not widely supported, or at least not 100%
 supported.  Therefore we use only some of the C99 features and explicitly
 disallow some (this will gradually be adjusted over time).
 
-Please don't make changes everywhere to use the C99 features, it causes merge
-problems for existing patches.  Only use them for new and changed code.
+Features not to be used ~
 
-Comments ~
+These C99 features are not to be used, because not enough compilers support
+them:
+- Variable length arrays (even in C11 this is an optional feature).
+- C99 _Bool and _Complex types.
+- "inline" (it's hardly ever needed, let the optimizer do its work)
+- flexible array members: Not supported by HP-UX C compiler (John Marriott)
 
-Traditionally Vim uses /* comments */.  We intend to keep it that way
-for file and function headers and larger blocks of code, E.g.:
+COMMENTS					       *style-comments*
+
+Comments are good, but there is also a danger of over-commenting. NEVER try to
+explain HOW your code works in a comment. Write the code so that the working
+is obvious.
```

Comment:
I am not sure this needs to be mentioned here.  This will lead to people not adding comments at all and stating that the code is obvious.  As Vim contributors have come and gone over the decades, it is very important to properly add comments describing the functionality.  Otherwise it is very difficult to understand and maintain the code.

Reply:
Fair.